src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -0,0 +1,1287 @@
+	case UFBX_BLEND_PIN_LIGHT: return "pinLight"_s;
+	case UFBX_BLEND_HARD_MIX: return "hardMix"_s;
+	case UFBX_BLEND_DIFFERENCE: return "difference"_s;
+	case UFBX_BLEND_EXCLUSION: return "exclusion"_s;
+	case UFBX_BLEND_SUBTRACT: return "subtract"_s;
+	case UFBX_BLEND_DIVIDE: return "divide"_s;
+	case UFBX_BLEND_HUE: return "hue"_s;
+	case UFBX_BLEND_SATURATION: return "saturation"_s;
+	case UFBX_BLEND_COLOR: return "color"_s;
+	case UFBX_BLEND_LUMINOSITY: return "luminosity"_s;
+	case UFBX_BLEND_OVERLAY: return "overlay"_s;
+    default:
+        Warning{} << "Unhandled blend mode" << Int(mode);
+        return {};
+    /* LCOV_EXCL_STOP */
+    }
+}
+
+ufbx_load_opts loadOptsFromConfiguration(Utility::ConfigurationGroup& conf) {
+    ufbx_load_opts opts = {};
+
+    const Long maxTemporaryMemory = conf.value<Long>("maxTemporaryMemory");
+    const Long maxResultMemory = conf.value<Long>("maxResultMemory");
+    const bool geometricTransformNodes = conf.value<bool>("geometricTransformNodes");
+    const bool preserveRootNode = conf.value<bool>("preserveRootNode");
+    const bool normalizeUnits = conf.value<bool>("normalizeUnits");
+
+    opts.generate_missing_normals = conf.value<bool>("generateMissingNormals");
+    opts.strict = conf.value<bool>("strict");
+    opts.disable_quirks = conf.value<bool>("disableQuirks");
+    opts.load_external_files = conf.value<bool>("loadExternalFiles");
+    opts.ignore_geometry = conf.value<bool>("ignoreGeometry");
+    opts.ignore_animation = conf.value<bool>("ignoreAnimation");
+    opts.ignore_embedded = conf.value<bool>("ignoreEmbedded");
+    opts.ignore_all_content = conf.value<bool>("ignoreAllContent");
+    opts.ignore_missing_external_files = true;
+
+    /* Substitute zero maximum memory to one, so that if the user computes the
+       maximum memory and ends up with zero it doesn't result in unlimited */
+    if(maxTemporaryMemory >= 0)
+        opts.temp_allocator.memory_limit = size_t(Utility::max(maxTemporaryMemory, Long(1)));
+    if(maxResultMemory >= 0)
+        opts.result_allocator.memory_limit = size_t(Utility::max(maxResultMemory, Long(1)));
+
+    /* By default FBX has cameras pointing at +X and lights at -Y, let ufbx
+       normalize those to what Magnum expects */
+    opts.target_light_axes = ufbx_axes_right_handed_y_up;
+    opts.target_camera_axes = ufbx_axes_right_handed_y_up;
+
+    /* @todo expose more of these as options? need to think of reasonable
+       defaults anyways, feels like ignoring geometry transform is not great */
+    if(preserveRootNode)
+        opts.space_conversion = UFBX_SPACE_CONVERSION_TRANSFORM_ROOT;
+    else
+        opts.space_conversion = UFBX_SPACE_CONVERSION_ADJUST_TRANSFORMS;
+
+    if(geometricTransformNodes)
+        opts.geometry_transform_handling = UFBX_GEOMETRY_TRANSFORM_HANDLING_HELPER_NODES;
+    else
+        opts.geometry_transform_handling = UFBX_GEOMETRY_TRANSFORM_HANDLING_PRESERVE; /* MODIFY_GEOMETRY? */
```

Comment:
There is now an option in ufbx to transform the geometry itself if possible even without helper nodes if we want to use it here, you can choose whether to create helper nodes for instanced meshes (`UFBX_GEOMETRY_TRANSFORM_HANDLING_MODIFY_GEOMETRY`) or just throw away the geometry transforms (`UFBX_GEOMETRY_TRANSFORM_HANDLING_MODIFY_GEOMETRY_NO_FALLBACK`).

Reply:
If the feature is there, it doesn't make sense to not use it :) 

I'm thinking of replacing the boolean `geometricTransformNodes` option with a `geometryTransformHandling` option that can be set to `helperNodes`, `preserve`, `modifyGeometry` or `modifyGeometryNoFallback` to cover these cases.

Hm, the `UFBX_GEOMETRY_TRANSFORM_HANDLING_MODIFY_GEOMETRY_NO_FALLBACK` option would interact with the `noInstanceMaterials` option [I suggested to remove in another comment](https://github.com/mosra/magnum-plugins/pull/133#discussion_r1045266792), right? As in, if you throw away geometry transforms, then the mesh has to be duplicated for about every node it's referenced from, thus implying `noInstanceMaterials` as well, right?

Reply:
With no fallback I mean *no* fallback, meaning it will just break for instanced meshes which have geometry transforms (but hey, that's what happens in Maya as well :D). Doing automatic duplication would be pretty painful given how FBX/ufbx works internally so probably not worth adding. An option would be to duplicate meshes on Magnum side but I think the helper nodes are unobtrusive to be enough.

Reply:
> it will just break for instanced meshes which have geometry transforms

:scream: sounds like an option I don't want, then :D (a "maya-compatible buggy loading" option, lol)

> Doing automatic duplication would be pretty painful

Yeah definitely don't do that. Magnum can do that itself if needed, like shown [in the `flattenMeshHierarchy()` code snippet here](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1SceneTools.html#a8b90ec6a17a8f3f3d1150dfdcedab40c), as well as any other hierarchy-cleaning operations.